observer/observer_test.go

```diff
@@ -0,0 +1,141 @@
+package observer
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	cmtjson "github.com/cometbft/cometbft/libs/json"
+	"github.com/cometbft/cometbft/libs/log"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	"github.com/cometbft/cometbft/rpc/jsonrpc/types"
+	cmttypes "github.com/cometbft/cometbft/types"
+	proto "github.com/cosmos/gogoproto/proto"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/require"
+
+	bridge "github.com/osmosis-labs/osmosis/v23/x/bridge/types"
+)
+
+const (
+	EventOutboundTransferType = "osmosis.bridge.v1beta1.EventOutboundTransfer"
+)
+
+var upgrader = websocket.Upgrader{}
+
+func readNewBlockEvent(path string) coretypes.ResultEvent {
+	dataStr, err := os.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	result := coretypes.ResultEvent{}
+	err = cmtjson.Unmarshal([]byte(dataStr), &result)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func readBlockResults(path string) coretypes.ResultBlockResults {
+	dataStr, err := os.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	result := coretypes.ResultBlockResults{}
+	err = json.Unmarshal([]byte(dataStr), &result)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func success(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
+		c, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			panic(err)
```

Comment:
Tests shouldn't panic, otherwise it's going to be a nightmare to work with them :) Good practice would be to call `require.NoError(t, err)` on every error. One way to achieve that in your case would be by using closures:

```go
func success(t *testing.T) http.HandlerFunc {
	return func(writer http.ResponseWriter, request *http.Request) {
		err := someMethod()
		require.NoError(t, err)

		// your code
	}
}
```

And then call it like this
```go
s := httptest.NewServer(success(t))
```